packages/patternfly-4/react-core/src/components/ApplicationLauncher/ApplicationLauncherItem.tsx

```diff
@@ -25,29 +27,58 @@ export interface ApplicationLauncherItemProps {
    * } />
    */
   component?: React.ReactNode;
+  /** Flag indicating if the item is favorited */
+  isFavorite?: boolean;
+  /** ID of the item. Required for tracking favorites. */
+  id?: string;
 }
 
 export const ApplicationLauncherItem: React.FunctionComponent<ApplicationLauncherItemProps & DropdownItemProps> = ({
   className = '',
+  id,
   children,
   icon = null,
   isExternal = false,
   href,
   tooltip = null,
   tooltipProps = null,
   component = 'a',
+  isFavorite = null,
   ...props
 }: ApplicationLauncherItemProps & DropdownItemProps) => (
   <ApplicationLauncherItemContext.Provider value={{ isExternal, icon }}>
-    <DropdownItem
-      component={component}
-      href={href || null}
-      className={css(isExternal && styles.modifiers.external, className)}
-      tooltip={tooltip}
-      tooltipProps={tooltipProps}
-      {...props}
-    >
-      {children && <ApplicationLauncherContent>{children}</ApplicationLauncherContent>}
-    </DropdownItem>
+    <ApplicationLauncherContext.Consumer>
+      {({ onFavorite }) => (
+        <DropdownItem
+          id={id}
+          component={component}
+          href={href || null}
+          className={css((isExternal || onFavorite) && styles.modifiers.link, className)}
+          listItemClassName={css(
+            (isExternal || onFavorite) && styles.appLauncherMenuWrapper,
+            isExternal && styles.modifiers.external,
+            isFavorite && styles.modifiers.favorite
+          )}
+          tooltip={tooltip}
+          tooltipProps={tooltipProps}
+          {...(isFavorite !== null && {
+            additionalChildren: (
+              <button
+                className={css(styles.appLauncherMenuItem, styles.modifiers.action)}
+                onClick={() => {
+                  onFavorite(id, isFavorite);
+                }}
+              >
+                <StarIcon />
+              </button>
+            )
+          })}
+          {...(isFavorite !== null && { 'aria-label': isFavorite ? 'starred' : 'not starred' })}
```

Comment:
should the consumer be able to change the aria-lable?  @jessiehuff

Reply:
I saw @mcoker's comment above.  We still need to be able to configure these strings for localization purposes.